bf2fc6cc711aee1a0c2a/kas-fleet-manager
# add kafka supported billing models to instance_types endpoint

## Description

Related issue: https://issues.redhat.com/browse/MGDSTRM-10001

This PR adds support to return the available KAS Fleet Manager billing models to the supported instance types endpoint(s) `/api/kafkas_mgmt/v1/instance_types/{cloud_provider}/{cloud_region}`. This is part of the work needed to support kafka long-lived eval instances functionality.

This PR deals with exposing the information to the endpoint, not making use of it yet. That will be performed in a future PR.

The KAS fleet manager billing models are defined with the attribute supported_billing_models which is a mandatory attribute for each instance type defined. Its type is a map.

Sample response:
```
  "instance_types": [
    {
      "id": "developer",
      "display_name": "Trial",
      "supported_billing_models": [
        {
          "id": "trial",
          "ams_resource": "rhosak",
          "ams_product": "RHOSAKTrial",
          "ams_billing_models": [
            "standard"
          ]
        }
      ],
     ...
  ...
...
```

## Verification Steps

1. Run kas fleet manager serve
1. Perform a http request to the instance_types endpoint: `/api/kafkas_mgmt/v1/instance_types/aws/us-east-1`
1. Verify that the results contain the `supported_billing_models` subsection accordingly to what's specified in the `config/kafka-instance-types-configuration.yaml` file.



## Checklist (Definition of Done)
